packages/back-end/src/routers/teams/teams.controller.ts

```diff
@@ -0,0 +1,262 @@
+  status: 200;
+  team: TeamInterface;
+};
+
+/**
+ * POST /teams
+ * Create a team resource
+ * @param req
+ * @param res
+ */
+export const postTeam = async (
+  req: CreateTeamRequest,
+  res: Response<CreateTeamResponse>
+) => {
+  const { org, userName } = getOrgFromReq(req);
+  const { name, description, permissions } = req.body;
+
+  req.checkPermissions("manageTeam");
+
+  const team = await createTeam({
+    name,
+    createdBy: userName,
+    description,
+    organization: org.id,
+    ...permissions,
+  });
+
+  await req.audit({
+    event: "team.create",
+    entity: {
+      object: "team",
+      id: team.id,
+      name: name,
+    },
+    details: auditDetailsCreate(team),
+  });
+
+  return res.status(200).json({
+    status: 200,
+    team,
+  });
+};
+
+// endregion POST /teams
+
+// region GET /teams
+
+type GetTeamsResponse = {
+  status: 200;
+  teams: TeamInterface[];
+};
+
+/**
+ * GET /teams
+ * Get all the teams for the authenticated user's organization
+ * @param req
+ * @param res
+ */
+export const getTeams = async (
+  req: AuthRequest,
+  res: Response<GetTeamsResponse>
+) => {
+  const { org } = getOrgFromReq(req);
+
+  req.checkPermissions("manageTeam");
+
+  const teams = await getTeamsForOrganization(org.id);
+
+  return res.status(200).json({
+    status: 200,
+    teams,
+  });
+};
+
+// endregion GET /teams
+
+// region GET /teams/:id
+
+type GetTeamResponse = {
+  status: 200 | 404;
+  team?: TeamInterface;
+  message?: string;
+};
+
+/**
+ * GET /teams/:id
+ * Get team document for the given id
+ * @param req
+ * @param res
+ */
+export const getTeamById = async (
+  req: AuthRequest<null, { id: string }>,
+  res: Response<GetTeamResponse>
+) => {
+  const { org } = getOrgFromReq(req);
+  const { id } = req.params;
+
+  req.checkPermissions("manageTeam");
+
+  const team = await findTeamById(id, org.id);
+  const members = org.members.filter((member) => member.teams?.includes(id));
+  const expandedMembers = await expandOrgMembers(members);
+
+  if (!team) {
+    return res.status(404).json({
+      status: 404,
+      message: "Cannot find team",
+    });
+  }
+
+  return res.status(200).json({
+    status: 200,
+    team: {
+      ...team,
+      members: expandedMembers,
+    },
+  });
+};
+
+// endregion GET /teams/:id
+
+// region PUT /teams/:id
+
+type PutTeamRequest = AuthRequest<
+  {
+    name: string;
+    createdBy: string;
+    description: string;
+    permissions: MemberRoleWithProjects;
+  },
+  { id: string }
+>;
+
+type PutTeamResponse = {
+  status: 200;
+};
+
+/**
+ * PUT /teams/:id
+ * Update team document for the given id
+ * @param req
+ * @param res
+ */
+export const updateTeam = async (
+  req: PutTeamRequest,
+  res: Response<PutTeamResponse>
+) => {
+  const { org } = getOrgFromReq(req);
+  const { name, description, permissions } = req.body;
+  const { id } = req.params;
+
+  req.checkPermissions("manageTeam");
+
+  const team = await findTeamById(id, org.id);
+
```

Comment:
handle updating members here?

Reply:
I think if we end up having a modal to add/edit a team and in that add/remove members, it makes sense to handle updating members here.

We'll probably also want to add logic to handle updating the teams a member is a part of within the `putMember` method.